OMCompiler/Compiler/BackEnd/Differentiate.mo

```diff
@@ -2060,13 +2060,49 @@ algorithm
       then
         (e, funcs);
 
+    // added for ticket #6068
+    // (p is a parameter)
+    // x^p
+    case DAE.BINARY(exp1 = e1,operator = DAE.POW(tp),exp2 = (e2 as DAE.CREF(componentRef = cr)))
+      equation
+        // this fails if it is not a parameter
+        etmp = match getParameterType(cr, inInputData)
+          case DAE.T_INTEGER()  then DAE.BINARY(e2, DAE.SUB(tp), DAE.ICONST(1));
+          else DAE.BINARY(e2, DAE.SUB(tp), DAE.RCONST(1.0));
+        end match;
+        (de1, funcs) = differentiateExp(e1, inDiffwrtCref, inInputData, inDiffType, inFunctionTree, maxIter);
+        e = DAE.BINARY(DAE.BINARY(e2,DAE.MUL(tp),
+                       DAE.BINARY(e1,DAE.POW(tp),etmp)),
+                       DAE.MUL(tp),de1);
+      then
+        (e, funcs);
```

Comment:
I think this case needs to consider `p=0` so that `der(x^0)=0`

Reply:
Yea sure, but `0 * x ^ (-1)` gets simplifie to `0` anyway.

There are some other things i have to fix here though. The DiffData does not contain the variables i need for this :(

Reply:
right, I'm blind :sweat_smile:

Reply:
I just realized that it is in general not possible here and also simplification will not remove it. Parameters are allowed to be changed after compilation. :)

Reply:
Absolutely, in that case an if construct like in the below case may be desired